src/hotspot/share/nmt/mallocTracker.cpp

```diff
@@ -91,8 +93,8 @@ size_t MallocMemorySnapshot::total_arena() const {
 void MallocMemorySnapshot::make_adjustment() {
   size_t arena_size = total_arena();
   int chunk_idx = NMTUtil::tag_to_index(mtChunk);
-  _malloc[chunk_idx].record_free(arena_size);
-  _all_mallocs.deallocate(arena_size);
+  _malloc[chunk_idx].record_free(arena_size, 0);
+  _all_mallocs.deallocate(arena_size, 0);
```

Comment:
Here also the `allocated` param is passed as `0`.  This makes the accounting incorrect. The requested changes but allocated not.